### PR TITLE
BHBC-1115

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

## Links to jira tickets

- [{BHBC-1115}](https://quartech.atlassian.net/browse/BHBC-1115)

## This PR contains the following changes

  - addition of survey_publish_history table
  - addition of survey_occurrence table
  - modification of api_delete_project to not allow deletions when published data exists

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Code

- [x] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [x] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [x] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [x] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [x] My code changes account for null/undefined values and handle errors appropriately
- [x] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [x] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: database/src/migrations/20210225205948_biohub_release.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import path from 'path';
 const DB_USER_API_PASS = process.env.DB_USER_API_PASS;
 const DB_USER_API = process.env.DB_USER_API;
 
+const DB_RELEASE = 'release.0.17'
```
Comment (sdevalapurkar) [inline database/src/migrations/20210225205948_biohub_release.ts]:
```suggestion
const DB_RELEASE = 'release.0.17';
```

**Review by sdevalapurkar — DISMISSED** [review]
code lgtm, minor nit comment

**Comment by sdevalapurkar** [comment]
also, may need to run `npm run format:fix` from `database` directory

File: database/src/migrations/20210225205948_biohub_release.ts (hunk 1 above)
Comment (cgarrettjones) [reply database/src/migrations/20210225205948_biohub_release.ts]: sorry i keep forgetting to run lint

**Review by sdevalapurkar — APPROVED** [review]
code lgtm

File: database/src/migrations/release.0.17/biohub.sql [hunk 2]
```diff
@@ -652,7 +652,15 @@ COMMENT ON COLUMN permit.update_user IS 'The id of the user who updated the reco
 ;
 COMMENT ON COLUMN permit.revision_count IS 'Revision count used for concurrency control.'
 ;
-COMMENT ON TABLE permit IS 'Provides a record of scientific permits.'
+COMMENT ON TABLE permit IS 'Provides a record of scientific permits. Note that permits are first class objects in the data model and do not require an association to either a project or survey. Additionally:
```
Comment (NickPhura) [inline database/src/migrations/release.0.17/biohub.sql:655]: nice!

**Pull request merged**